cudax/include/cuda/experimental/__detail/config.cuh

```diff
@@ -21,37 +21,35 @@
 #  pragma system_header
 #endif // no system header
 
-// Debuggers do not step into functions marked with __attribute__((__artificial__)).
-// This is useful for small wrapper functions that just dispatch to other functions and
-// that are inlined into the caller.
-#if _CCCL_HAS_ATTRIBUTE(__artificial__) && !_CCCL_HAS_CUDA_COMPILER()
-#  define _CUDAX_ARTIFICIAL __attribute__((__artificial__))
-#else // ^^^ _CCCL_HAS_ATTRIBUTE(__artificial__) ^^^ / vvv !_CCCL_HAS_ATTRIBUTE(__artificial__) vvv
-#  define _CUDAX_ARTIFICIAL
-#endif // !_CCCL_HAS_ATTRIBUTE(__artificial__)
-
-// _CCCL_HIDE_FROM_ABI and _CCCL_FORCEINLINE cannot be used together because they both
-// try to add `inline` to the function declaration. In cudax we solve this problem with
-// two attributes:
+// _CCCL_HIDE_FROM_ABI and _CCCL_FORCEINLINE cannot be used together because they both try
+// to add `inline` to the function declaration. In cudax we solve this problem with two
+// attributes:
 // - `_CUDAX_API` declares the function host/device and hides the symbol from the ABI
-// - `_CUDAX_TRIVIAL_API` does the same while also forcing inlining and hiding the function from debuggers
+// - `_CUDAX_TRIVIAL_API` does the same while also forcing inlining and hiding the
+//   function from debuggers
 #define _CUDAX_API        _CCCL_HOST_DEVICE _CCCL_VISIBILITY_HIDDEN _CCCL_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
 #define _CUDAX_HOST_API   _CCCL_HOST _CCCL_VISIBILITY_HIDDEN _CCCL_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
 #define _CUDAX_DEVICE_API _CCCL_DEVICE _CCCL_VISIBILITY_HIDDEN _CCCL_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
```

Comment:
I was thinking about this a bit further and I believe we should just introduce `_CCCL_API` and friends